src/consensus/vote.cpp

```diff
@@ -570,6 +614,82 @@ std::string VoteManager::getJsonStr(std::vector<Vote> const& votes) {
   return ptroot.toStyledString();
 }
 
+std::vector<std::shared_ptr<Vote>> VoteManager::getProposalVotes(uint64_t pbft_round) {
+  std::vector<std::shared_ptr<Vote>> proposal_votes;
+
+  sharedLock_ lock(verified_votes_access_);
+  for (auto const& voted_value : verified_votes_[pbft_round][1]) {
+    for (auto const& v : voted_value.second) {
+      proposal_votes.emplace_back(std::make_shared<Vote>(v.second));
```

Comment:
I know we were talking a lot about shared pointers and references but I am not sure if you completely understand what we want to achieve by that. What do you think this vector of shared_ptrs solves in terms of efficiency if you are still making a copy of vote here std::make_shared<Vote>(v.second) as you stored Votes objects in verified_votes_ ?

Reply:
Sorry, I don't understand how to do here.
I thought std::make_shared<Vote>(v.second) is makeing a shared pointer to the vote object. 
Could you please give me an example? How should I do here? Thanks